crates/proof/src/eigenda_witness_provider.rs

```diff
@@ -0,0 +1,64 @@
+use alloy_primitives::FixedBytes;
+use async_trait::async_trait;
+use hokulea_compute_proof::compute_kzg_proof;
+use hokulea_eigenda::{AltDACommitment, EigenDABlobProvider, EigenDAVersionedCert};
+use hokulea_proof::cert_validity::CertValidity;
+use hokulea_proof::eigenda_blob_witness::EigenDABlobWitnessData;
+use rust_kzg_bn254_primitives::blob::Blob;
+use std::sync::{Arc, Mutex};
+
+/// This is a wrapper around OracleEigenDAProvider, with
+/// additional functionalities to generate eigenda witness
+/// which is KZG proof on the FS point out of the blob itself.
+/// OracleEigenDAWitnessProvider is only inteneded to be used outside
+/// FPVM or ZKVM. Its sole purpose is to generate KZG proof at the
+/// client side
+#[derive(Debug, Clone)]
+pub struct OracleEigenDAWitnessProvider<T: EigenDABlobProvider> {
```

Comment:
should we call it `OracleEigenDAKZGProofProvider` then? Or `KZGWitnessProvider`?